core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -337,13 +337,14 @@ oppia.controller('StateResponses', [
               $scope.trainingDataFeedback = feedbackHtml;
               $scope.trainingDataOutcomeDest = dest;
 
-              var answerGroupIndex =
-                classificationResult.answerGroupIndex;
+              var classificationCategorization = (
+                classificationResult.classificationCategorization);
+              var answerGroupIndex = classificationResult.answerGroupIndex;
               var ruleIndex = classificationResult.ruleIndex;
-              if (answerGroupIndex !==
-                _state.interaction.answerGroups.length &&
-                  _state.interaction.answerGroups[answerGroupIndex]
-                    .rules[ruleIndex].type !== RULE_TYPE_CLASSIFIER) {
+
+              if (classificationCategorization !==
```

Comment:
I think it would be more helpful to use an === here (i.e. say what it should be rather than what it isn't).

Reply:
Done